core/blockchain.go

```diff
@@ -2355,6 +2360,9 @@ func (bc *BlockChain) insertSideChain(block *types.Block, it *insertIterator) (i
 	for i := len(hashes) - 1; i >= 0; i-- {
 		// Append the next block to our batch
 		block := bc.GetBlock(hashes[i], numbers[i])
+		if block == nil {
+			continue
```

Comment:
If it happens, continue will cause chaos. Is it possible to happen?

Reply:
It won't happen because delete block that block has state
above 'hashes' is that no state block hash, so will access ancient data
add 'if block == nil ... continue’ judgement , only consider code rigor.

Reply:
I see, I suggest log fatal here, to avoid unexpected consequence.